AbstractMethodError when trying to get serializable proxy

If you have a situation where a proxy is being serialized and the session is no longer available or open, the initializer tries to wrap with a `SerializableProxy`. 

However, the dynamic `GroovyAwareJavassistLazyInitializer` fails to implement the `serializableProxy` proxy abstract method. The method is created at:

https://github.com/grails/gorm-hibernate5/blob/3c981581cc35ed6aba43f7fa980ccf1331dddaea/grails-datastore-gorm-hibernate5/src/main/groovy/org/grails/orm/hibernate/proxy/ProxyFactorySupport.groovy#L82

But `initializerClass.addMethod(serializableProxyMethod)` is never called to add it.

This results in an `AbstractMethodError` when the method is called in the `BasicLazyInitializer`.

The real exception is also hidden by the wrapping of the exception message only at:

https://github.com/grails/gorm-hibernate5/blob/3c981581cc35ed6aba43f7fa980ccf1331dddaea/grails-datastore-gorm-hibernate5/src/main/groovy/org/grails/orm/hibernate/proxy/ProxyFactorySupport.groovy#L116

Since the AbstractMethodError has no message body, you end up with an exception like:
```
Caused by: java.lang.Exception: null
        at org.grails.orm.hibernate.proxy.GroovyAwareJavassistLazyInitializer.invoke(GroovyAwareJavassistLazyInitializer.java)
        at com.triu.contract.CBFOutputFile_$$_jvst300_103.writeReplace(CBFOutputFile_$$_jvst300_103.java)
        at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
        at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
        at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
        at java.lang.reflect.Method.invoke(Method.java:498)
        at java.io.ObjectStreamClass.invokeWriteReplace(ObjectStreamClass.java:1230)
        ... 99 common frames omitted
```

If you remove that try catch, you will see the real `AbstractMethodException`.

This is a major problem for our application because we are using Spring Session with JDBC support so the flash is being serialized. In some cases, errors or exceptions in the flash have uninitialized proxies that need to be serialized and will fail in these cases.

I have a PR to fix that i'll submit shortly.

--- a/grails-datastore-gorm-hibernate5/src/main/groovy/org/grails/orm/hibernate/proxy/ProxyFactorySupport.groovy
+++ b/grails-datastore-gorm-hibernate5/src/main/groovy/org/grails/orm/hibernate/proxy/ProxyFactorySupport.groovy
@@ -92,6 +92,7 @@ class ProxyFactorySupport {
                 setIdentifierMethod,
                 componentIdType);       
 }""")
+            initializerClass.addMethod(serializableProxyMethod)
 
             // Add method: Object invoke(Object, Method, Method, Object[])
             CtClass methodType = getClassFromPool(pool, Method.name)

